raiden/network/pathfinding.py

```diff
@@ -193,63 +237,57 @@ def create_current_iou(
         our_address: typing.Address,
         privkey: bytes,
         block_number: typing.BlockNumber,
+        offered_fee: int = None,
+        scrap_existing_iou: bool = False,
 ) -> typing.Dict[str, typing.Any]:
+
     url = config['pathfinding_service_address']
-    latest_iou = get_pfs_iou(url, token_network_address)
+
+    latest_iou = None
+    if not scrap_existing_iou:
+        latest_iou = get_pfs_iou(url, token_network_address)
+
     if latest_iou is None:
         return make_iou(
             config=config,
             our_address=our_address,
             privkey=privkey,
             block_number=block_number,
+            offered_fee=offered_fee,
         )
     else:
-        added_amount = config['pathfinding_max_fee']
+        added_amount = offered_fee or config['pathfinding_max_fee']
         return update_iou(iou=latest_iou, privkey=privkey, added_amount=added_amount)
 
 
-def query_paths(
-        service_config: typing.Dict[str, typing.Any],
-        our_address: typing.Address,
-        privkey: bytes,
-        current_block_number: typing.BlockNumber,
-        token_network_address: typing.Union[typing.TokenNetworkAddress, typing.TokenNetworkID],
-        route_from: typing.InitiatorAddress,
-        route_to: typing.TargetAddress,
-        value: typing.TokenAmount,
-) -> typing.List[typing.Dict[str, typing.Any]]:
-    max_paths = service_config['pathfinding_max_paths']
-    url = service_config['pathfinding_service_address']
-    payload = {'from': route_from, 'to': route_to, 'value': value, 'max_paths': max_paths}
-
-    payload.update(create_current_iou(
-        config=service_config,
-        token_network_address=token_network_address,
-        our_address=our_address,
-        privkey=privkey,
-        block_number=current_block_number,
-    ))
-
+def post_pfs_paths(url, token_network_address, payload):
     try:
         response = requests.post(
             f'{url}/api/v1/{to_checksum_address(token_network_address)}/paths',
             data=payload,
             timeout=DEFAULT_HTTP_REQUEST_TIMEOUT,
         )
-    except requests.RequestException:
+    except requests.RequestException as e:
         raise ServiceRequestFailed(
-            'Could not connect to Pathfinding Service',
+            f'Could not connect to Pathfinding Service ({e})',
```

Comment:
I believe you have to do `{str(e)}` as trying to output the exception as-is would break logging for some reason.

Reply:
True, I changed it.